src/bitmap/store/bitmap_store.rs

```diff
@@ -309,10 +318,14 @@ impl<B: Borrow<[u64; BITMAP_LENGTH]>> Iterator for BitmapIter<B> {
         loop {
             if self.value == 0 {
                 self.key += 1;
-                if self.key >= BITMAP_LENGTH {
+                let cmp = self.key.cmp(&self.key_back);
+                self.value = if cmp == Ordering::Less {
+                    unsafe { *self.bits.borrow().get_unchecked(self.key) }
+                } else if cmp == Ordering::Equal {
+                    self.value_back
+                } else {
                     return None;
```

Comment:
```suggestion
                self.value = match if self.key.cmp(&self.key_back) {
                    Ordering::Less => unsafe { *self.bits.borrow().get_unchecked(self.key),
                    Ordering::Equal => self.value_back,
                    Ordering::Greater => return None,
```

Reply:
🙅 See comment above re: match arm reordering

Reply:
Oh. The comment from this suggested edit didn't make it into the code.